assets/src/components/v2/disruption_diagram/disruption_diagram.tsx

```diff
@@ -99,175 +119,91 @@ const endLabelIDMap: { [labelID: string]: string[] } = {
 };
 
 interface IconProps {
-  x: number;
-  className?: string;
+  iconSize: number;
 }
 
-// Translate by: width of the icon + stroke-width
-const CurrentStopIconRedLine: ComponentType<IconProps> = ({ x }) => {
-  const iconWidth = 52;
-  const strokeWidth = 4;
-
-  return (
-    <>
-      <path
-        transform={`translate(${x - (iconWidth + strokeWidth) / 2} ${
-          -(iconWidth - strokeWidth) / 3
-        })`}
-        d="M32.6512 3.92661C30.0824 1.3578 25.9176 1.3578 23.3488 3.92661L3.92661 23.3488C1.3578 25.9176 1.3578 30.0824 3.92661 32.6512L23.3488 52.0734C25.9176 54.6422 30.0824 54.6422 32.6512 52.0734L52.0734 32.6512C54.6422 30.0824 54.6422 25.9176 52.0734 23.3488L32.6512 3.92661Z"
-        className="middle-slot__background--red"
-        stroke="#E6E4E1"
-        strokeWidth={strokeWidth}
-        strokeLinejoin="round"
-      />
-      <path
-        transform={`translate(${x - (iconWidth + strokeWidth) / 2} ${
-          -(iconWidth - strokeWidth) / 3
-        })`}
-        fillRule="evenodd"
-        clipRule="evenodd"
-        d="M15.4855 29.219C14.7045 28.438 14.7045 27.1717 15.4855 26.3906L26.3906 15.4855C27.1717 14.7045 28.438 14.7045 29.219 15.4855L40.1242 26.3906C40.9052 27.1717 40.9052 28.438 40.1241 29.219L29.219 40.1242C28.438 40.9052 27.1717 40.9052 26.3906 40.1241L15.4855 29.219Z"
-        fill="white"
-      />
-    </>
-  );
-};
-
-// Translate by: width of the icon + stroke-width
-const CurrentStopIcon: ComponentType<IconProps> = ({ x }) => {
-  const iconWidth = 52;
-  const strokeWidth = 4;
-
-  return (
-    <>
-      <path
-        transform={`translate(${x - (iconWidth + strokeWidth) / 2} ${
-          -(iconWidth - strokeWidth) / 3
-        })`}
-        d="M3.15665 25.2076C1.61445 26.7498 1.61445 29.2502 3.15665 30.7924L25.2076 52.8434C26.7498 54.3856 29.2502 54.3856 30.7924 52.8434L52.8434 30.7924C54.3856 29.2502 54.3856 26.7498 52.8434 25.2076L30.7924 3.15668C29.2502 1.61448 26.7498 1.61448 25.2076 3.15668L3.15665 25.2076Z"
-        fill="#EE2E24"
-        stroke="#E6E4E1"
-        strokeWidth={strokeWidth}
-      />
-    </>
-  );
-};
+interface EndpointProps {
+  className: string;
+}
 
-const CurrentStopIconEndpointRedLine: ComponentType<IconProps> = ({ x }) => {
-  const iconWidth = 64;
+// Non-circle icons are translated by their top-left corner, while circles
+// are translated by their center-point. So to position these non-circles,
+// translate is x shifted by half the width of the icon, and y is shifted up half its
+// iconsize and half the thickness of the line diagram itself
+const translateNonCircleIcon = (iconSize: number) => (
+  `translate(${-iconSize / 2} -${(iconSize-LINE_HEIGHT) / 2})`
```

Comment:
Just noticed this and it's super minor, but the x translate has the negative applied to the number, while the y translate adds the negative as part of the string. Could you apply it the same way for both, for consistency? No strong preference on one way or the other.